### DISCUSSION TITLE
Why useMutation resets status and isPending when user navigate between routes

### DISCUSSION THREAD

**Discussion by ZeroProger** [TanStack/query#7480] [Q&A] [answered]
Need help with `useMutation`. 
`isPending` resets to `false` and `status` resets to `idle` when i very fast change page: 
from `/servers/[serverId]/overview` to `/servers/[serverId]/players` and back to `/servers/[serverId]/overview`.

In devtools network this request still pending, but `status` resets to `idle` and `isPending` resets to `false`.

If i stay on`/servers/[serverId]/overview` page (where i start request) `status` and `isPending` works correct.

I use `"next": "^13.5.2"` with app router.
I use `"@tanstack/react-query": "^5.38.0"`
I can`t disable request send button, if page changed - button is not disabled, is page not changed - button disabled correctly.

how i use it:
```ts
export function useStopServerMutation() {
	const serverHash = useStore($serverHash)
	const queryClient = useQueryClient()

	return useMutation({
		mutationFn: stopServer,
		onSettled: async (response, error) => {
			if (!response) return

			if (response.success) {
				await queryClient.invalidateQueries({ queryKey: [ReactQueryKeys.server] })
				await queryClient.invalidateQueries({
					queryKey: [ReactQueryKeys.serverMainInfo, serverHash],
				})
			}
		},
	})
}
```
```tsx
export function StopServer() {
	const serverHash = useStore($serverHash)

	const { isPending, mutateAsync } = useStopServerMutation()

	const handleClick = () => {
		mutateAsync({ gameServerHash: serverHash! })
	}

	return (
		<Button
			className="text-xl flex flex-row items-center flex-nowrap gap-2"
			onClick={handleClick}
			disabled={isPending}
			variant="ghost"
		>
			<Square size={24} className="text-destructive fill-destructive" />
			Остановить
		</Button>
	)
}
```

**Comment by TkDodo** [comment, marked as answer]
Mutations don't share state like Queries do. Whenever you mount a component, it gets a new instance of a Mutation. This makes sure you can trigger the same mutation multiple times.

So if you navigate away and back, `useMutation` is "fresh" and doesn't know anything about any mutation fired in the past.

You can use `useMutationState` to read the state of all mutations in the cache.
